src/Meshes/mesh.ts

```diff
@@ -4372,13 +4376,23 @@ export class Mesh extends AbstractMesh implements IGetSetVerticesData {
             vertexData.transform(wm);
             return vertexData;
         };
-        const vertexData = getVertexDataFromMesh(source).merge(meshes.slice(1).map((mesh) => getVertexDataFromMesh(mesh)), allow32BitsIndices);
+
+        const sourceVertexData = getVertexDataFromMesh(source);
+        yield;
+
+        const meshVertexDatas = new Array<VertexData>(meshes.length - 1);
+        for (let i = 1; i < meshes.length; i++) {
+            meshVertexDatas[i - 1] = getVertexDataFromMesh(meshes[i]);
+            yield;
+        }
+
+        const vertexData: VertexData = yield* sourceVertexData._mergeCoroutine(meshVertexDatas, allow32BitsIndices);
 
         if (!meshSubclass) {
             meshSubclass = new Mesh(source.name + "_merged", source.getScene());
         }
 
-        (<VertexData>vertexData).applyToMesh(meshSubclass);
+        yield* vertexData._applyToCoroutine(meshSubclass);

```

Comment:
I'm afraid that could not work with IE and unfortunately this is pretty fundamental in the engine

Did you try it in IE?

Reply:
When you say it is fundamental to the engine, what do you mean? As far as I can tell, this is the only place `applyToMesh` is used, and this is in the context of `MergeMeshes`, which is only used in a few places like `hemisphereBuilder` and `sceneOptimizer`. These seem like not more minor aspects of BJS.

I did not try it in IE. Is it generator functions in general that don't work in IE, or just the `yield*` part to yield another generator?

Reply:
yield and generator will not compile on IE
By "fundamental to the engine", I mean it is a core function that is used by several 1st party partners who (unfortunately) still needs to support IE

Reply:
I don't see any `yield` statements in the transpiled JS code, I guess because of the ES version being targeted by BJS, so presumably this won't break IE. I dev on a MacBook, but I can try to get a BJS dev environment setup on a Windows machine to test.

Reply:
![image](https://user-images.githubusercontent.com/16655165/141691652-8c94b99e-1f46-47fc-88bf-523c78c498da.png)

Some of the TS allowed constructs despite not emitting code relying on the feature might require other unsupported feature which is the case for instance with the async await syntax and the promise polyfill. I am not sure the required symbols are supported on IE.

Reply:
I removed both the `yield*` and the `downlevelIteration` as I realized I could just manually iterate over the nested coroutine which is more verbose but functions identically.